joblib/test/test_parallel.py

```diff
@@ -1228,6 +1228,35 @@ def test_parallel_return_order_with_return_as_generator_parameter(n_jobs):
     assert all(v == r for v, r in zip(input_list, result))
 
 
+def _sqrt_with_delay(e, delay):
+    if delay:
+        sleep(30)
+    return sqrt(e)
+
+
+@pytest.mark.parametrize('n_jobs', [2, 4])
+# NB: for this test to work, the backend must be allowed to process tasks
+# concurrently, so at least two jobs with a non-sequential backend are
+# mandatory.
+@with_multiprocessing
+@parametrize('backend', set(RETURN_GENERATOR_BACKENDS) - {"sequential"})
+def test_parallel_unordered_generator_returns_fastest_first(backend, n_jobs):
+    # This test submits 10 tasks, but the second task is super slow. This test
+    # checks that the 9 other tasks return before the slow task is done, when
+    # `return_as` parameter is set to `'generator_unordered'`
+    result = Parallel(n_jobs=n_jobs, return_as="generator_unordered",
+                      backend=backend)(
+        delayed(_sqrt_with_delay)(i**2, (i == 1)) for i in range(10))
+
+    quickly_returned = sorted(next(result) for _ in range(9))
+
+    expected_quickly_returned = [0] + list(range(2, 10))
+
+    assert all(
+        v == r for v, r in zip(expected_quickly_returned, quickly_returned)
+    )
+
```

Comment:
:thinking: will investigate that. Maybe forcing gc collection for pypy like we do in some of the other tests willl be enough indeed.

Reply:
My point is not about GC: here the long task is still running in the background, so there would be nothing to GC.

My point is about killing the worker without waiting for the task to complete. Otherwise debugging other tests can be very confusing if there are still long running tasks from previous independent tests being processed in the background.